Tests/RugbyTests/Core/Dependencies/VersionParserTests.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  VersionParserTests.swift
+//  
+//
+//  Created by Marcel Canhisares on 2023-09-30.
+//
+@testable import Rugby
+import XCTest
+
+final class VersionParserTests: XCTestCase {
+    private var parser: VersionParser!
+    
+    override func setUp() {
+        super.setUp()
+        parser = VersionParser()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        parser = nil
+    }
+}
+
+extension VersionParserTests {
+
+    func testParseValidVersion() throws {
+        let version = "1.0.0"
+        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.prod, build: 0)
+        let parsedVersion = try self.parser.parse(version)
+        XCTAssertEqual(ghVersion.major, parsedVersion.major)
+        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertEqual(ghVersion.build, parsedVersion.build)
```

Comment:
Please, use `// Assert & Act` sections similar to other tests files.
Also, you can skip `self` here.
```suggestion
        let version = "1.0.0"
        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.prod, build: 0)
        
        // Act
        let parsedVersion = try self.parser.parse(version)
        
        // Assert
        XCTAssertEqual(ghVersion.major, parsedVersion.major)
        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
        XCTAssertEqual(ghVersion.build, parsedVersion.build)
```